### PR TITLE
RSDK-8265 - Share connection in signaling answerer

### PR DESCRIPTION
**Pull request by cheukt**
mostly a prefactor so that it will be easier to refactor to use a passed in connection later on.

main behavior change is that the reconnect logic got tossed, idea being that any direct grpc connections will reconnect on their own anyway

Tested a few cases to ensure no regressions and confirmed that we can connect to the robot through WebRTC
* Long idle connection (over 1hr)
* Switching network interfaces (changing wifis/to my phone hotspot)
* Losing internet

Also ran tests with RDK to ensure those still passed

### CODE REVIEW

**Review by abe-winter — APPROVED** [review]
I'm not competent in this codebase so wait for second review if you want actual :eye: , but lgtm

File: rpc/wrtc_signaling_answerer.go
Comment (abe-winter) [reply rpc/wrtc_signaling_answerer.go:123]: actually ignore this, resolving

File: rpc/wrtc_signaling_answerer.go
Comment (cheukt) [reply rpc/wrtc_signaling_answerer.go]: ya it's the same, was lazy

File: rpc/dial_test.go
Comment (cheukt) [inline rpc/dial_test.go:1036]: Start and Serve calls ans.Start(), so having both actually causes a goroutine leak

File: rpc/wrtc_signaling_answerer.go
Comment (cheukt) [reply rpc/wrtc_signaling_answerer.go:113]: good to know, but ya, not sure how it'll interact

File: rpc/wrtc_signaling_answerer.go
Comment (benjirewis) [inline rpc/wrtc_signaling_answerer.go:95]:
```suggestion
// via Stop. Start cannot be called more than once before a Stop().
```

File: rpc/wrtc_signaling_answerer.go
Comment (benjirewis) [inline rpc/wrtc_signaling_answerer.go:101]: [q] Are you now using the `answerWorkers` wait group to keep track of the connection establishment goroutine + the eventual, two answer workers? That seems fine to me, but wanted to check.

File: rpc/wrtc_signaling_answerer.go
Comment (benjirewis) [reply rpc/wrtc_signaling_answerer.go:113]: I realize you copied this code from existing, but could we make 10s a constant and call it `answererConnectTimeout` or something?

**Review by benjirewis — COMMENTED** [review]
Nice! Solid improvement.

File: rpc/wrtc_signaling_answerer.go
Comment (cheukt) [reply rpc/wrtc_signaling_answerer.go]: yep, updated some var names as well to reflect

**Review by benjirewis — APPROVED** [review]
LGTM! Thanks 🧑‍🔧 .

**Pull request merged**
